release-notes/swan-lake-2201.1.0-release-note.md

```diff
@@ -30,9 +30,364 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
+
+The `lang.decimal:quantize()` function has been introduced to control the precision of decimal values, which returns a value equal to the first operand after rounding and having the exponent of the second operand.
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+    io:println(decimal:quantize(123.123, 1.0)); // 123.1
+    io:println(decimal:quantize(123.123, 1.00)); // 123.12
+    io:println(decimal:quantize(123.123, 1.000)); // 123.123
+}
+```
+If the length of the coefficient after the quantize operation would be greater than the precision, the function results in a panic.
+
+```ballerina
+public function main() {
+    decimal _ = decimal:quantize(123.1233, 1E-36); // results in a panic
+}
+```
+
+###### New `lang.float:toFixedString()` and `lang.float:toExpString()` functions
+
+Two new functions namely `lang.float:toFixedString()` and `lang.float:toExpString()` have been introduced to get the string representation of a `float` value in fixed-point notation and scientific notation respectively. Both the functions allow you to specify the number of digits required after the decimal point.
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+    string a = float:toFixedString(5.7, 16); 
+    io:println(a); // 5.7000000000000002
+    string b = float:toFixedString(5.7, 2); 
+    io:println(b); // 5.70
+    
+    float f1 = -45362.12334;
+    string c = float:toExpString(f1, 16); 
+    io:println(c); // -4.5362123339999998e+4
+    string d = float:toExpString(f1, 2); 
+    io:println(d); // -4.54e+4
+}
+```
+
+###### New `lang.string:padStart()`, `lang.string:padEnd()`, and `lang.string:padZero()` functions
+
+The `lang.string:padStart()`, `lang.string:padEnd()`, and `lang.string:padZero()` functions have been introduced to add padding in strings. `lang.string:padStart()` adds padding to the start of a string. `lang.string:padEnd()` adds padding to the end of a string. `lang.string:padZero()` pads a string with zeros.
+
+```ballerina
+public function main() {
+    io:println("abc".padStart(5, "#")); // "##abc"
+    io:println("abc".padStart(5)); // "  abc"
+
+    io:println("abc".padEnd(5, "#")); // "abc##"
+    io:println("abc".padEnd(5)); // "abc  "
+
+    io:println("123".padZero(5)); // "00123"
+    io:println("123".padZero(5, "#")); // "##123"
+}
+```
+
+#### Improvements
+
+##### Disallow inferring array length in contexts that are not permitted
+
+Inferring array length has been restricted to list constructors in variable and constant declarations. Moreover, only the first dimension can be inferred in multidimensional arrays.
+
+```ballerina
+int[*] x1 = [1, 2]; // Supported.
+
+int[2] y = [1, 2];
+int[*] x2 = y; // Not supported. Requires a list constructor to infer the array length.
+
+int[*][2] x3 = [[1, 2], [1, 2]]; // Supported.
+int[*][*] x4 = [[1, 2], [1, 2]]; // Not supported. Only the first dimension can be inferred.
+```
+
+##### Revamped `lang.float:round` langlib function
```

Comment:
```suggestion
##### Revamped `lang.float:round` langlib function
```
Is `lang lib` one word or 2 words? We have it as 2 words in the title and both need to be consistent.

Reply:
In title we have lang library and langlib as one word in short form.

Reply:
Then, the shorten for should be `lang lib`.